src/figuresClasses.ts

```diff
@@ -2,10 +1,94 @@
+'useStrict';
 
-export class Triangle implements Figure {}
+enum Shapes {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+enum Colors {
+  red = 'red',
+  green = 'green',
+  blue = 'blue',
+}
+
+export interface Figure {
+  shape: Shapes;
+  color: Colors;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shapes = Shapes.Triangle;
+
+  constructor(
+    public readonly color: Colors,
+    public readonly a: number,
+    public readonly b: number,
+    public readonly c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0 || !this.isValidTriangle()) {
+      throw new Error(`Sides ${a}, ${b}, and ${c} cannot form a triangle`);
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
+
+  private isValidTriangle(): boolean {
+    const sides = [this.a, this.b, this.c];
+    const highestNumber = Math.max(...sides);
+    const sumOfSides = sides.reduce(
+      (accumulator: number, currentValue: number) => accumulator + currentValue,
+    );
```

Comment:
The sumOfSides calculation includes all sides, which is correct, but the comparison in line 49 should ensure that the sum of the two smaller sides is greater than the largest side. The current logic checks if the largest side is less than the sum of all sides minus the largest side, which is equivalent but less clear. Consider using a more straightforward check: `return this.a + this.b > this.c && this.a + this.c > this.b && this.b + this.c > this.a;`.